src/index.ts

```diff
@@ -54,6 +70,93 @@ const unknownSerialize = (
   return O.tryCatch(() => Sum.serialize(x as any))
 }
 
+/**
+ * Without an assertion, encapsulated here, `Object.keys` will err on the side
+ * of caution and widen `Array<Tag<A>>` to `Array<string>`.
+ */
+const unsafeGetTags: <A extends Sum.AnyMember>(
+  x: Record<Tag<A>, unknown>,
+) => Array<Tag<A>> = Object.keys
+
+/**
+ * Unlike `t.union` this requires only at least one member rather than two.
+ */
+const union1 = <A extends [t.Mixed, ...Array<t.Mixed>]>(
+  xs: A,
+  name = "union1",
+): t.UnionType<A, t.TypeOf<A[number]>, t.OutputOf<A[number]>> =>
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  xs[1] === undefined
+    ? (new t.UnionType(
+        name,
+        xs[0].is,
+        xs[0].validate,
+        xs[0].encode,
+        xs,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ) as any)
+    : t.union(xs as unknown as [t.Mixed, t.Mixed, ...Array<t.Mixed>], name)
+
+/**
+ * Fold a `MemberCodecs` struct to a union codec from left to right.
+ *
+ * This function isn't well typed, however absent this abstraction the call
+ * sites would still be riddled in assertions, making this tradeoff a win on
+ * ergonomic grounds.
+ */
+const foldToUnion =
+  <A extends Sum.AnyMember>(f: (x: [Tag<A>, t.Type<Value<A>>]) => t.Mixed) =>
+  (
+    cs: MemberCodecs<A>,
+    name = "Sum union",
+  ): t.UnionType<[t.Mixed, ...Array<t.Mixed>]> =>
+    pipe(
+      cs,
+      // NB `R.toArray` doesn't preserve object order.
+      Object.entries,
```

Comment:
Seems I am little bit old school and come from the unordered approach of properties, but I learned something new: https://2ality.com/2015/10/property-traversal-order-es6.html

Still don't know what to think about it because now I can't sort my key alphabetically or in a semantically sense as I still need to keep in mind that this influences the order.

Reply:
This only affects externally tagged and untagged representations; adjacently tagged and internally tagged members share the same tag key, making their encoded representations mutually exclusive.

We can't strictly decode rejecting on any foreign property as most APIs return properties we don't care about, and will do so with new ones without considering it a breaking change.

We could ask for an `Ord` or something to make the precedence more explicit, but it won't usually be relevant, and it's documented. `t.union` behaves the same way:

```ts
const foo = t.strict({ foo: t.null })
const bar = t.strict({ bar: t.null })

const foobar = t.union([foo, bar])
const barfoo = t.union([bar, foo])

const compat: t.TypeOf<typeof foo> & t.TypeOf<typeof bar> = { foo: null, bar: null }

expect(foobar.decode(compat)).toStrictEqual(E.right({ foo: null }))
expect(foobar.decode(compat)).not.toStrictEqual(E.right({ bar: null }))

expect(barfoo.decode(compat)).toStrictEqual(E.right({ bar: null }))
expect(barfoo.decode(compat)).not.toStrictEqual(E.right({ foo: null }))
```